### PR TITLE
feat(components/lookup): tokenize search styles

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2868/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2868/storybooks/ag-grid-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2868/storybooks/data-manager-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2868/storybooks/lookup-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2868/storybooks/phone-field-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2868/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2868/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2868/playground/)

File: libs/components/lookup/src/lib/modules/search/search.component.scss [hunk 1]
```diff
@@ -89,16 +68,41 @@ sky-search {
...
+            var(--sky-color-border-selected)
+          )
+      );
+      color: var(
+        --sky-override-search-applied-color,
+        var(--sky-color-icon-action)
+      );
+      padding: var(
+        --sky-override-search-mobile-button-padding,
+        var(--sky-space-inset-balanced-s)
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/lookup/src/lib/modules/search/search.component.scss]: The spec seems to call for `space.inset.pillarbox.1_2.{r,l}.m` and `space.inset.pillarbox.1_2.{t,b}.m` not `space.inset.balanced.s`. Is the current token correct?

File: libs/components/lookup/src/lib/modules/search/search.component.scss [hunk 2]
```diff
@@ -123,37 +127,77 @@ sky-search {
...
+        --sky-override-search-background-color,
+        var(--sky-color-background-action-secondary-base)
+      );
     }
 
     .sky-input-box .sky-input-group-btn .sky-btn.sky-search-btn {
-      width: 40px;
+      height: var(
+        --sky-override-search-button-height-width,
+        calc(
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/lookup/src/lib/modules/search/search.component.scss:195]: Is there a reason that these calculations are slightly different than what is in the spec?

```
height: calc(var(--sky-line_height-body-m) + calc(var(--sky-spacing-inset-balanced-s) * 2));
```

File: libs/components/lookup/src/lib/modules/search/search.component.scss [hunk 3]
```diff
@@ -173,36 +217,37 @@ sky-search {
...
       .sky-input-box-icon-inset-left-wrapper {
         .sky-input-box-icon-inset-left {
-          padding: 0 10px;
+          padding: 0 var(--sky-space-gap-icon-m);
         }
       }
 
       .sky-input-group-icon {
         .sky-icon {
           font-size: 17.3333px; // Override input-box's size with the old font-awesome size.
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/lookup/src/lib/modules/search/search.component.scss]: Should this be `var(--sky-size-icon-s)` for v2 with overrides?

**Review by Blackbaud-TrevorBurch — COMMENTED** [review]
I think we also need to look at how toolbars are affecting the search component. For example, the mobile search button loses its soft color when search text is applied and search is collapsed. Also the collapsed search button has different padding.

File: libs/components/lookup/src/lib/modules/search/search.component.scss (hunk 2 above)
Comment (Blackbaud-ErikaMcVey) [reply libs/components/lookup/src/lib/modules/search/search.component.scss:195]: yes, line height needs to be multiplied by font size to get an actual height value there, and the `body-m` was incorrect, input boxes set a different line height

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**